### ISSUE TITLE
Xiaomi Binary sensor displaying incorrect state in 0.84.2 and 0.84.3

### ISSUE THREAD

**Issue by aidbish** [home-assistant/home-assistant-polymer#2351]
Home Assistant release with the issue:
0.84.2 and 0.84.3
Frontend JavaScript version: latest

Last working Home Assistant release (if known):
0.83

Browser and Operating System:

Description of problem:
Xiaomi binary door/window sensors now display state as on/off instead of open/close periodically. If rebooting, they sometimes show correctly only after a while they revert to on/off. Have customized entities to the correct devices classes to see if it fixes the issue but still happens

No errors are shown in the logs

This is a follow on to https://github.com/home-assistant/home-assistant-polymer/issues/2326

This update did not fix the issue and the states consistently display on/off instead of open/close
i have modified the device class for Door or windows where appropriate for the device

**Comment by VDRainer** [comment]
Seeing this also in mqtt binary_sensor.

**Comment by balloob** [comment]
Duplicate of #2351

**Comment by aidbish** [comment]
hi @balloob this is issue 2351

**Comment by balloob** [comment]
I meant #2354
